src/java/com/puppetlabs/puppetserver/ShellUtils.java

```diff
@@ -39,7 +56,13 @@ private static ExecutionResult executeExecutor(CommandLine commandLine,
         //    PipedInputStream stdoutInputStream = new PipedInputStream(stdoutOutputStream);
         // but this requires that the input stream be read on a different thread
         // than this one. this is currently out of scope.
-        ByteArrayOutputStream stdoutOutputStream = new ByteArrayOutputStream();
+        ByteArrayOutputStream stdoutOutputStream;
+        if (options.getCombineStdoutStderr()) {
+            log.debug("Combining STDOUT/STDERR for external command '" + commandLine.toString() + "'");
+            stdoutOutputStream = errStream;
```

Comment:
Is passing the same stream object for both `out` and `err` into `PumpStreamHandler` thread-safe?